pkg/vm/engine/disttae/logtailreplay/partition_state.go

```diff
@@ -68,6 +69,10 @@ type PartitionState struct {
 	// should have been in the Partition structure, but doing that requires much more codes changes
 	// so just put it here.
 	shared *sharedStates
+
+	// blocks deleted before minTS is hard deleted.
+	// partition state can't serve txn with snapshotTS less than minTS
+	minTS atomic.Pointer[types.TS]
```

Comment:
this field should be types.TS and never be modified. The PartitionState is immutable.